### PR TITLE
Add SSL loadbalancer article

### PR DESCRIPTION
**Pull request by RobertJansen1**

### CODE REVIEW

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+
+## Testing the loadbalancer.
+Now that the load balancer is created, we can test it
+
+**Step 1**: Create an A record in DNS for the domain to point to the floating IP address
+[managing DNS records](
+{{ '/articles/managing-dns-records' | relative_url }})
+
+
+**Step 2**: Await the update of the load balancer's Operating Status to ONLINE and the DNS to
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:353]: Please add two spaces here to add an enter

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+Finalize the deployment and start testing
+([Testing the loadbalancer](#testing-the-loadbalancer))
+
+
+---
+
+## Testing the loadbalancer.
+Now that the load balancer is created, we can test it
+
+**Step 1**: Create an A record in DNS for the domain to point to the floating IP address
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:348]: Please add two spaces here to add an enter

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+[managing DNS records](
+{{ '/articles/managing-dns-records' | relative_url }})
+
+
+**Step 2**: Await the update of the load balancer's Operating Status to ONLINE and the DNS to
+propagate. Once this status is achieved, navigate to `https://DomainName` in your web browser to 
+witness your load balancer functioning.  
+
+**Step 3**: Verify the SSL of the load balancer to have your URL checked with the
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:357]: Please add two spaces here to add an enter

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+Create the health monitor
+
+```bash
+openstack loadbalancer healthmonitor create "${pool_uuid}" --name "webserver-healthmonitor-http" \
+--type HTTP --delay 5 --timeout 5 --max-retries 3
+
+```
+
+**Step 6**
+
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:328]: Please remove the enter whitespace

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+--listener "${listener_uuid}"
+```
+Make sure to save the returned `id` as variable, we need that later
+```bash
+pool_uuid="uuid"
+```
+
+
+**Step 5**
+
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:317]: Please remove the whitespace
image

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+--default-tls-container-ref "${octavia_certificate_url}" 
+```
+
+Make sure to save the returned `id` as variable, we need that later
+```bash
+listener_uuid=uuid
+```
+
+**Step 4**
+
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:302]: Please remove the whitespace

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 7]
```diff
@@ -0,0 +1,361 @@
...
+Now we can create the loadbalancer. We will create a loadbalancer with a listener, a pool and a
+healthmonitor. It is only possible to create the loadbalancer through 
+
+**Prerequisites**
+We should already have a bash variable set with the certificate location
+```bash
+octavia_certificate_url="https://keymanager.domain.tld:/v1/containers/uuid"
+```
+**Step 1**
+
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:269]: Please remove the whitespace

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+**Step 18
+Finalize the deployment and start testing
+([Testing the loadbalancer](#testing-the-loadbalancer))
+
+## Creating the loadbalancer using the OpenStack CLI
+Now we can create the loadbalancer. We will create a loadbalancer with a listener, a pool and a
+healthmonitor. It is only possible to create the loadbalancer through 
+
+**Prerequisites**
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:263]: Please add two spaces here to add an a line break

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 9]
```diff
@@ -0,0 +1,361 @@
...
+--payload="$(base64 < "$certificate")" --expiration "${expiration_date}"
+```
+Make sure to save the returned secret_href as variable, we need that later
+```bash
+octavia_certificate_url="https://keymanager.domain.tld:/v1/secrets/uuid"
+```
+
+---
+
+## Creating the loadbalancer
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:176]: Maybe add line breaks behind the steps so the style of the steps is the same throughout the article (Optional)

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 10]
```diff
@@ -0,0 +1,361 @@
...
+**Step 2**
+Convert the certificates to a pkcs12 certificate (skip this if you already have a pkcs12 encoded
+file with all required certificates):
+```bash
+openssl pkcs12 -export -inkey private.key -in certificate.pem -certfile intermediate.pem \
+-passout pass: -out complete.p12
+```
+
+**Step 3**
+
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:157]: Please remove the whitespace

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 11]
```diff
@@ -0,0 +1,361 @@
...
+**Step 1**  
+First make sure you have setup the OpenStack CLI and that you are able to
+execute commands using the `openstack` command. For more information please
+refer to the
+[Using the OpenStack CLI article](
+    {{ '/articles/using-the-cli-linux' | relative_url }}
+).
+
+**Step 2**
+
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:53]: Please remove the whitespace

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 12]
```diff
@@ -0,0 +1,361 @@
...
+openstack secret store --name="${name}" -t 'application/octet-stream' -e 'base64' \
+--payload="$(base64 < "$certificate")" --expiration $(date --date="$(openssl x509 -enddate -noout \
+-in "$certificate"|cut -d= -f 2)" --iso-8601)
+```
+Make sure to save the returned secret_href as variable, we need that later
+```bash
+certificate_url="https://keymanager.domain.tld:/v1/secrets/uuid"
+```
+
+**Step 3**
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:68]: Please remove the whitespace

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 13]
```diff
@@ -0,0 +1,361 @@
...
+--payload $(read -sp "Password: ";echo ${REPLY})
+```
+Answer the password question and press enter
+
+Make sure to save the returned secret_href as variable, we need that later
+```bash
+passphrase_url="https://keymanager.domain.tld:/v1/secrets/uuid"
+```
+
+**Step 4**
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:83]: Please add two spaces here to add an a line break

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 14]
```diff
@@ -0,0 +1,361 @@
...
+certificate=private.key
+openstack secret store --name="${name}" -t 'application/octet-stream' -e 'base64' \
+--payload="$(base64 < "$certificate")"
+```
+Make sure to save the returned secret_href as variable, we need that later
+```bash
+private_key_url="https://keymanager.domain.tld:/v1/secrets/uuid"
+```
+
+**Step 5**
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:97]: Please add two spaces here to add an a line break

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 15]
```diff
@@ -0,0 +1,361 @@
...
+openstack secret store --name="${name}" -t 'application/octet-stream' -e 'base64' \
+--payload="$(base64 < "$certificate")" --expiration $(date --date="$(openssl x509 -enddate -noout \
+-in "$certificate"|cut -d= -f 2)" --iso-8601)
+```
+Make sure to save the returned secret_href as variable, we need that later
+```bash
+intermediates_url="https://keymanager.domain.tld:/v1/secrets/uuid"
+```
+
+**Step 6**
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:112]: Please add two spaces here to add an a line break

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 16]
```diff
@@ -0,0 +1,361 @@
...
+refer to the
+[Using the OpenStack CLI article](
+    {{ '/articles/using-the-cli-linux' | relative_url }}
+).
+
+**Step 2**
+
+Store the certificate in barbican
+```bash
+certificate=certificate.pem
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:56]: Please add some comments and/or some whitespace so the article is more readable 

An example of this you can find at the bottom of the [Resize Volume](https://cloudtutorials.eu/articles/resize-volume) article (Check the last code block)

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 17]
```diff
@@ -0,0 +1,361 @@
...
+Make sure to save the returned secret_href as variable, we need that later
+```bash
+certificate_url="https://keymanager.domain.tld:/v1/secrets/uuid"
+```
+
+**Step 3**
+Store the passhprase for the private key in barbican.
+
+```bash
+name="${domain}_passphrase"
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:72]: Please add some comments and/or some whitespace so the article is more readable 

An example of this you can find at the bottom of the [Resize Volume](https://cloudtutorials.eu/articles/resize-volume) article (Check the last code block)

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 18]
```diff
@@ -0,0 +1,361 @@
...
+Make sure to save the returned secret_href as variable, we need that later
+```bash
+private_key_url="https://keymanager.domain.tld:/v1/secrets/uuid"
+```
+
+**Step 5**
+Store all intermediate certificates in barbican
+
+```bash
+certificate=intermediate.pem
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:101]: Please add some comments and/or some whitespace so the article is more readable 

An example of this you can find at the bottom of the [Resize Volume](https://cloudtutorials.eu/articles/resize-volume) article (Check the last code block)

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 19]
```diff
@@ -0,0 +1,361 @@
...
+
+**Step 5**
+
+
+Create the health monitor
+
+```bash
+openstack loadbalancer healthmonitor create "${pool_uuid}" --name "webserver-healthmonitor-http" \
+--type HTTP --delay 5 --timeout 5 --max-retries 3
+
```
Comment (houtknots) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:324]: Please remove the whitespace below the command

**Review by houtknots — CHANGES_REQUESTED** [review]
Needs some change to make the article better readable

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 20]
```diff
@@ -0,0 +1,361 @@
...
+The alternative way to store a certificate to use with an Octavia loadbalancer is to create a single
+pkcs12 file with all certificates without password protection.
+The downside of this approach is we need to temporarily store the unencrypted private key for the
+load balancer on the OpenStack CLI server. The benefit is we can select the certificate in Horizon.
+
+
+**Prerequisites**
+ - The server certificate, intermediate certificates and private key are stored in the proper order in
+a single file on the OpenStack CLI server named 'certificate.pem'
+ - openssl tooling installed on the OpenStack CLI server
```
Comment (bkranendonk) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:138]: OpenSSL

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 21]
```diff
@@ -0,0 +1,361 @@
...
+```bash
+intermediates_url="https://keymanager.domain.tld:/v1/secrets/uuid"
+```
+
+**Step 6**
+Create a certificate container containing the certificate, all intermediates
+the passphrase and the private key.
+
+```bash
+name="${domain}_container"
```
Comment (bkranendonk) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:117]: Maybe code comments here

File: _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md [hunk 22]
```diff
@@ -0,0 +1,361 @@
...
+healthmonitor.
+
+**Step 1**: Navigate to the `Network` tab and select `Load Balancers`.  
+**Step 2**: Initiate the process by clicking on the `Create Load Balancer` button.  
+**Step 3**: Enter details in the following fields:  
+* **Name**: webserver-loadbalancer
+* **IP Address**: Leave empty for now
+* **Description**: Loadbalancer for our webservers
+* **Availability Zone**: Leave empty or choose an availability zone to your liking
+* **Flavor**: Choose a flavor to your liking for this tutorial we use the Medium flavor
```
Comment (bkranendonk) [inline _posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md:193]: Interpunction

**Review by houtknots — APPROVED** [review]

**Pull request merged**